config/v3_1/config.go

```diff
@@ -56,6 +59,28 @@ func Parse(rawConfig []byte) (types.Config, report.Report, error) {
 	return config, rpt, nil
 }
 
-func isEmpty(userdata []byte) bool {
-	return len(userdata) == 0
+// ParseConfig parses the raw config of version 3.1.0 or lesser
+// into a 3.1 types.Config struct and generates a report of any
+// errors, warnings,info, and deprecations it encountered
+func ParseConfig(raw []byte) (types.Config, report.Report, error) {
```

Comment:
The difference between `Parse` and `ParseConfig` isn't clear from the names.  We can't change `Parse` because it's part of the stable API.  Maybe `ParseCompatibleVersion`?